src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,211 @@
+                registrationService.register(user),
+                "There must RuntimeException throw "
+                        + "when NULL AGE field in User registration");
+        assertEquals(expectedSize, Storage.people.size(),
+                "Storage size not changed properly");
+    }
+
+    @Test
+    void registering_underValidAgeUser_notOK() {
+        user.setAge(15);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user),
+                "There must RuntimeException throw "
+                        + "when UNDER VALID AGE field in User registration");
+        assertEquals(expectedSize, Storage.people.size(),
+                "Storage size not changed properly");
+    }
+
+    @Test
+    void registering_negativeAgeUser_notOK() {
+        user.setAge(-23);
+        assertThrows(RuntimeException.class, () -> registrationService.register(user),
+                "There must RuntimeException throw "
+                        + "when NEGATIVE AGE field in User registration");
+        assertEquals(expectedSize, Storage.people.size(),
+                "Storage, size not changed properly");
+    }
+
+    @Test
+    void registering_existentUser_notOK() {
+        long oldId = user.getId();
+        User actual = registrationService.register(user);
+        assertNotNull(actual,"Returned User cannot be null");
+        assertEquals(actual, user, "Registration method "
+                + "must return registered User");
+        assertNotEquals(oldId, user.getId(), "User Id must should be changed "
+                + "due to registration");
+        expectedSize++;
+
+        assertThrows(RuntimeException.class, () -> registrationService.register(user),
+                "There must RuntimeException throw "
+                        + "when try REGISTER SAME User");
+        assertEquals(expectedSize, Storage.people.size(),
+                "Storage size not changed");
+    }
+
+    @Test
+    void registering_existentLoginUser_notOK() {
+        User[] users = new User[2];
+        users[0] = user.createUser(750_001L, DEFAULT_LOGIN, "password1", DEFAULT_ADULT_AGE);
+        users[1] = user.createUser(750_002L, DEFAULT_LOGIN, "password2", DEFAULT_ADULT_AGE);
+
+        long oldId = users[0].getId();
+        User actual = registrationService.register(users[0]);
+        assertNotNull(actual, "Returned User cannot be null");
+        assertEquals(actual, users[0], "Registration method "
+                + "must return registered User");
+        assertNotEquals(oldId, users[0].getId(), "User Id must should be changed "
+                + "due to registration");
+        expectedSize++;
+
+        assertThrows(RuntimeException.class, () -> registrationService.register(users[1]),
+                "There must RuntimeException throw "
+                        + "when try register User with SAME LOGIN");
+        assertEquals(expectedSize, Storage.people.size(), "Storage size not changed properly");
```

Comment:
you have way too much setup and asserts in all of your test methods. For example, this scenario is adequately tested by just two lines something like this (if your default user is setup with valid values): 
        registrationService.register(user);
        assertThrows(InvalidUserException.class, () ->
                registrationService.register(user), "User already exists!"